src/Core/Content/Seo/SeoUrlGenerator.php

```diff
@@ -129,18 +149,28 @@ private function getSeoPathInfo(SeoUrlMapping $mapping, SeoUrlRouteConfig $confi
         }
     }
 
-    private function setTwigTemplate(SeoUrlRouteConfig $config, string $template): void
+    private function createTwigTemplate(SeoUrlRouteConfig $config, string $template): ?Environment
     {
+        $twig = $this->seoUrlTwigFactory->createTwigEnvironment($this->slugify, $this->twigExtensions);
+
         $template = "{% autoescape '" . self::ESCAPE_SLUGIFY . "' %}$template{% endautoescape %}";
-        $this->twig->setLoader(new ArrayLoader(['template' => $template]));
+        $twig->setLoader(new ArrayLoader(['template' => $template]));
 
         try {
-            $this->twig->loadTemplate($this->twig->getTemplateClass('template'), 'template');
+            $twig->loadTemplate($twig->getTemplateClass('template'), 'template');
         } catch (SyntaxError $syntaxError) {
+            $this->logger->critical('Error initializing SEO URL template', [
+                'exception' => $syntaxError,
```

Comment:
log the $mapping. Otherwise this will be pretty useless as you can't identifiy which seo url is failing here

Reply:
Good hint!